src/core/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,453 @@
+  }
+  return !!date2 && adjustedHours === date2.getHours();
+};
+
+const isSameMinute = (date1: Date, date2: Date | undefined) => {
+  return !!date2 && date1.getMinutes() === date2.getMinutes();
+};
+
+const isSameSecond = (date1: Date, date2: Date | undefined) => {
+  return !!date2 && date1.getSeconds() === date2.getSeconds();
+};
+
+const isSamePeriod = (period: PeriodType, date: Date | undefined) => {
+  return (
+    !!date && (period === 'AM' ? date.getHours() < 12 : date.getHours() >= 12)
+  );
+};
+
+const formatHourFrom12 = (hour: number, period: PeriodType | undefined) => {
+  const adjustedHour = hour % 12;
+  return period === 'PM' ? adjustedHour + 12 : adjustedHour;
+};
+
+const setHours = (hour: number, date: Date) => {
+  return new Date(
+    date.getFullYear(),
+    date.getMonth(),
+    date.getDate(),
+    hour,
+    date.getMinutes(),
+    date.getSeconds(),
+  );
+};
+
+export type PeriodType = 'AM' | 'PM';
+
+export type TimePickerProps = {
+  /**
+   * Selected date. Only time is used from Date object.
+   */
+  date?: Date;
+  /**
+   * Callback when time is changed.
+   */
+  onChange?: (date: Date) => void;
+  /**
+   * Format of the time. 12h or 24h.
+   * @default false
+   */
+  use12Hours?: boolean;
+  /**
+   * Precision of the time.
+   * @default 'minutes'
+   */
+  precision?: 'hours' | 'minutes' | 'seconds';
+  /**
+   * Change step of the hours displayed.
+   * @default 1
+   */
+  hourStep?: number;
+  /**
+   * Change step of the minutes displayed.
+   * @default 1
+   */
+  minuteStep?: number;
+  /**
+   * Change step of the seconds displayed.
+   * @default 1
+   */
+  secondStep?: number;
+  /**
+   * Set focus on hour.
+   * @default false
+   */
+  setFocusHour?: boolean;
+  /**
+   * Custom hour cell renderer.
+   * @default (date: Date) => <>{date.getHours()}</>
+   */
+  hourRenderer?: (date: Date) => JSX.Element;
+  /**
+   * Custom minute cell renderer.
+   * @default (date: Date) => <>{date.getMinutes()}</>
+   */
+  minuteRenderer?: (date: Date) => JSX.Element;
+  /**
+   * Custom second cell renderer.
+   * @default (date: Date) => <>{date.getSeconds()}</>
+   */
+  secondRenderer?: (date: Date) => JSX.Element;
+  /**
+   * Custom period cell renderer.
+   * @default (period: PeriodType) => <>{period}</>
+   */
+  periodRenderer?: (period: PeriodType) => JSX.Element;
+} & StylingProps;
+
+/**
+ * Time picker component
+ * @example
+ * <TimePicker date={new Date()} onChange={(e) => console.log('New time value: ' + e)} />
+ */
+export const TimePicker = (props: TimePickerProps): JSX.Element => {
+  const {
+    date,
+    onChange,
+    use12Hours = false,
+    precision = 'minutes',
+    hourStep = 1,
+    minuteStep = 1,
+    secondStep = 1,
+    setFocusHour = false,
+    hourRenderer = (date: Date) => <>{date.getHours()}</>,
+    minuteRenderer = (date: Date) => <>{date.getMinutes()}</>,
+    secondRenderer = (date: Date) => <>{date.getSeconds()}</>,
```

Comment:
All these renderers, I wonder if they should default to two digits?

Maybe using [`toLocaleString`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString)
```tsx
date.getHours().toLocaleString(undefined, { minimumIntegerDigits: 2 })
```

Just weird seeing 24 hour format without preceding zeroes. 08:23 makes more sense than 8:23 (without AM).

Reply:
using now for hours, mins and secs...